### ISSUE TITLE
[a11y-defect-2]: Memorable date has 4 unresolved major issues 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#61363]
## Point of contact

joshkimux

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

The memorable date component on the add new issue page has [three major defects which we've documented for the design system team here.](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1593)
> The most critical of the three is on VoiceOver iOS, where an error is drawn every time the user swipes right to the next field.

I'm uncertain if the 4th issue (see below) should be resolved through DST or is something we control (I could use @Mottie 's help here). I haven't added it to the DST memorable date bug ticket yet as a result. If we determine it should be there, I'll add it in.
> When a user attempts to add an issue, focus does not go to the first fieldset with an error.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/a4d9e06e-d03f-4df3-95d2-45ceb4e9bedb

## Reproduction steps

For bug #4 related to focus management

1. Log in to NoD as user 236
2. Turn on VoiceOver on Safari
3. On the contestable issues page, add an issue
4. When adding the issue, input 10, 10, 1993
5. This should create an error message
6. Attempt to add the issue
7. Confirm that focus is not managed to the memorable date component

## Proposed solution or next steps for the fourth issue

Focus management should be consistent with other parts of the form where we send focus to the first component with an issue. 

Small caveat-- I recently discovered an iOS bug with how errors are read out (I'm currently guessing that due to the several `span` tags we have in our error messages, only the first of the error before the `span`s will read out), which I'll need to do a bit more digging into. This may change my recommendation and require a DST ticket (I'll write this if I can confirm if it's an issue with some other specialists).

### Next steps

- [ ] Engineer to confirm if issue 4 is an issue we control, or something specific to the memorable date that should be handled by the design system team
- [ ] If this is something we can control, implement proposed solution of sending focus to the first fieldset with an error on the page upon submit
- [ ] If this is something we can't control, @joshkimux to write a ticket for the design system

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
Yes, our team is responsible for this 4th issue fix.

**Comment by Mottie** [comment]
Fixes submitted for:
- [Focus management for date errors](https://github.com/department-of-veterans-affairs/vets-website/pull/24826)
- [Bug with older dates](https://github.com/department-of-veterans-affairs/component-library/pull/749)
- [Error messaging does not address February](https://github.com/department-of-veterans-affairs/component-library/pull/750)

Bugs not yet fixed (from [DST ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1593), I probably won't tackle these):
- NVDA bug (Entering Jan, JAN or January) - enhancement is out of scope of the work I'm doing
- Error onblur between fields - the web component loses focus completely when "done" is used. Not sure how to fix this.

**Comment by HeatherWidmont** [comment]
We will keep this open until @joshkimux returns from PTO and can take a look at this next week.

**Comment by joshkimux** [comment]
I see this, and will be taking a look as soon as my body recovers this week ❤️

**Comment by joshkimux** [comment]
Confirming the following after testing on iOS VoiceOver:
- [x] Focus management for date errors
- [x] Bug with older dates
- [x] February bug

Let's close out this ticket knowing the other issues are out of @Mottie 's scope 💙 
This is awesome work, thank you so much!
